grpc-server-runtime/build.gradle

```diff
@@ -8,6 +8,7 @@ dependencies {
     api libs.micronaut.runtime
     api libs.managed.grpc.protobuf
     api libs.managed.grpc.stub
+    api libs.managed.grpc.services
```

Comment:
should this be `compileOnly` with `@Requires` defined in the bean? That way you could add this dependency to enable/disable health checks

Reply:
Does requires work with POJOs?  HealthCheckManager isn't a bean

Reply:
Hmmmm...  That then fails when it tries to build a server in the runtime-client tests...

I wonder if `implementation` is the way forward 🤔

Reply:
Ahhh but then, there's no point in the @Requires 😔

Reply:
Yeah, but the client-runtime tests (ie: HelloWordGrpcSpec) failed with

```
  io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type  [io.micronaut.grpc.server.GrpcEmbeddedServer]

  Message: Error loading bean [io.micronaut.grpc.server.GrpcServerBuilder]: Could not initialize class io.micronaut.grpc.server.$GrpcServerBuilder$Definition
  Path Taken: new GrpcEmbeddedServer(ApplicationContext applicationContext,ApplicationConfiguration applicationConfiguration,GrpcServerConfiguration grpcServerConfiguration,ServerBuilder serverBuilder,ApplicationEventPublisher eventPublisher,ComputeInstanceMetadataResolver computeInstanceMetadataResolver,List metadataContributors) --> new GrpcEmbeddedServer(ApplicationContext applicationContext,ApplicationConfiguration applicationConfiguration,GrpcServerConfiguration grpcServerConfiguration,[ServerBuilder serverBuilder],ApplicationEventPublisher eventPublisher,ComputeInstanceMetadataResolver computeInstanceMetadataResolver,List metadataContributors)
```

Which worries me that it's required on the classpath for anyone using gRPC

Maybe I'm doing something wrong

Reply:
Ok, instead of adding a `HealthCheckManager` to the constructor of `GrpcServerBuilder`, I can add a container class that holds the `HealthCheckManager` singleton, then inject that in its place.  Then the introspection is still valid

https://github.com/micronaut-projects/micronaut-grpc/pull/516/commits/c6cf4f7804f0b874a85387038fbfec6686e5a70b